src/python/pants/core/goals/fix.py

```diff
@@ -237,37 +251,44 @@ def _print_results(
         console.print_stderr(f"{sigil} {tool} {status}.")
 
 
-@goal_rule
-async def fix(
-    console: Console,
+_CoreRequestType = TypeVar("_CoreRequestType", bound=FixRequest)
+_TargetPartitioner = TypeVar("_TargetPartitioner", bound=FixTargetsRequest.PartitionRequest)
+_FilePartitioner = TypeVar("_FilePartitioner", bound=FixFilesRequest.PartitionRequest)
+_GoalT = TypeVar("_GoalT", bound=Goal)
+
+
+@rule_helper
+async def _do_fix(
+    core_request_types: Iterable[type[_CoreRequestType]],
+    target_partitioners: Iterable[type[_TargetPartitioner]],
+    file_partitioners: Iterable[type[_FilePartitioner]],
+    goal_cls: type[_GoalT],
+    subsystem: GoalSubsystem,
```

Comment:
I'm sure there's a way to type hint `subsystem` so mypy knows there is a `batch_size` on it..?

Reply:
maybe, but the underlying helper doesn't use that if we could `Protocol` it :/
I'll stew on it...

Reply:
Oh Protocols are fun!